Client/Frontend/Browser/Playlist/PlaylistInfo.swift

```diff
@@ -0,0 +1,81 @@
+// Copyright 2020 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import WebKit
+import Shared
+
+private let log = Logger.browserLogger
+
+struct PlaylistInfo: Codable {
+    let name: String
+    let src: String
+    let pageSrc: String
+    let pageTitle: String
+    let mimeType: String
+    let duration: Float
+    let detected: Bool
+    let dateAdded: Date
+    
+    init(item: PlaylistItem) {
+        self.name = item.name ?? ""
+        self.src = item.mediaSrc ?? ""
+        self.pageSrc = item.pageSrc ?? ""
+        self.pageTitle = item.pageTitle ?? ""
+        self.mimeType = item.mimeType ?? ""
+        self.duration = item.duration
+        self.dateAdded = item.dateAdded ?? Date()
+        self.detected = false
+    }
+    
+    init(name: String, src: String, pageSrc: String, pageTitle: String, mimeType: String, duration: Float, detected: Bool, dateAdded: Date) {
+        self.name = name
+        self.src = src
+        self.pageSrc = pageSrc
+        self.pageTitle = pageTitle
+        self.mimeType = mimeType
+        self.duration = duration
+        self.detected = detected
+        self.dateAdded = dateAdded
+    }
+    
+    init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.name = (try? container.decode(String.self, forKey: .name)) ?? ""
+        self.src = (try? container.decode(String.self, forKey: .src)) ?? ""
+        self.pageSrc = (try? container.decode(String.self, forKey: .pageSrc)) ?? ""
+        self.pageTitle = (try? container.decode(String.self, forKey: .pageTitle)) ?? ""
+        self.mimeType = (try? container.decode(String.self, forKey: .mimeType)) ?? ""
+        self.duration = (try? container.decode(Float.self, forKey: .duration)) ?? 0.0
+        self.detected = (try? container.decode(Bool.self, forKey: .detected)) ?? false
```

Comment:
Use `decodeIfPresent` instead?

Also why are you absorbing any thrown errors? You should just use `try`

Reply:
No because I want to decode each field. If a field isn't valid, doesn't mean the entire object is invalid. Some videos/websites don't have a mime-type. I don't want to make every single field optional and I don't want to fail initialization of the object because of that one missing field.

I modified it to throw on the mandatory fields and left the rest optional with default value.